src/gui/src/renderThread.cpp

```diff
@@ -979,6 +1042,224 @@ void RenderThread::drawLayer(QPainter* painter,
              layer_timer);
 }
 
+// Draw the region of the block.  Depth is not yet used but
+// is there for hierarchical design support.
+void RenderThread::drawBlockConcurrent(QPainter* painter,
+                             dbBlock* block,
+                             QVector<QRect>& subRects,
+                             QVector<QFuture<QImage>>& futures,
+                             int threadCount,
+                             const QRect& draw_bounds,
+                             int depth)
+{
+  const Rect dbu_bounds = viewer_->screenToDBU(draw_bounds);
+
+  utl::Timer timer;
+
+  utl::Timer manufacturing_grid_timer;
+  const int instance_limit = viewer_->instanceSizeLimit();
+
+  GuiPainter gui_painter(painter,
+                         viewer_->options_,
+                         dbu_bounds,
+                         viewer_->pixels_per_dbu_,
+                         block->getDbUnitsPerMicron());
+
+  // Draw die area, if set
+  painter->setPen(QPen(Qt::gray, 0));
+  painter->setBrush(QBrush());
+  Rect bbox = block->getDieArea();
+  if (bbox.area() > 0) {
+    painter->drawRect(bbox.xMin(), bbox.yMin(), bbox.dx(), bbox.dy());
+  }
+
+  drawManufacturingGrid(painter, dbu_bounds);
+  debugPrint(logger_,
+             GUI,
+             "draw",
+             1,
+             "manufacturing grid {}",
+             manufacturing_grid_timer);
+
+  /* Search instances */
+  utl::Timer inst_timer;
+  std::vector<dbInst*> insts;
+  std::vector<std::vector<dbInst*>> sub_insts;
+  insts.reserve(10000);
+  {
+    QVector<QFuture<std::vector<dbInst*>>> inst_futures;
+    for (const auto& subRect : subRects) {
+      const Rect dbu_subrect = viewer_->screenToDBU(subRect);
+      inst_futures.push_back(
+          QtConcurrent::run([=]() {
+            auto sub_inst_range = viewer_->search_.searchInsts(block,
```

Comment:
I have not studied this PR, I just want to say that I found the searching for triangles to be what dominated performance.

Any ideas on how this part can be parallelized?

I think @maliberty said something to the effect that making rtree run with multithreading is "not easy"...